solidity/contracts/PhasedEscrow.sol

```diff
@@ -70,26 +70,127 @@ contract PhasedEscrow is Ownable {
     }
 }
 
-interface ICurveRewards {
-    function notifyRewardAmount(uint256 amount) external;
+/// @title BatchedPhasedEscrow
+/// @notice A token holder contract allowing contract owner to approve a set of
+///         beneficiaries of tokens held by the contract, to appoint a separate
+///         drawee role, and allowing that drawee to withdraw tokens to approved
+///         beneficiaries in phases.
+contract BatchedPhasedEscrow is Ownable {
```

Comment:
Should this contract implement `IBeneficiaryContract` in case we want to transfer funds between escrows?

Reply:
I do not think so but let's validate what I say three times, please. When we'll be transferring from `PhasedEscrow` to `BatchedPhasedEscrow`, we'll set `StakerRewardsBeneficiary` on `PhasedEscrow` that will point to `BatchedPhasedEscrow` as a `_stakerRewards` beneficiary. This scenario is covered in `"can be funded from PhasedEscrow contract"` in unit tests. 

If we ever decide to transfer funds out of `BatchedPhasedEscrow`, we'll approve beneficiary using the standard mechanism... and that's it.

Did I miss something?

Reply:
Yeah, you're right. So transfer between escrows requires a beneficiary contract deployed. I'm not sure why we don't do this transfer directly by implementing just one `__escrowSentTokens` function?

Reply:
Same for `LPRewards`, why don't we implement `__escrowSentTokens` there instead of implementing an intermediary beneficiary contract?

Reply:
I think the original intention was to do not leak any details about how do we handle transfers between escrow into the final recipient. In this design, a beneficiary with `__escrowSentTokens` performs the role similar to what adapters do in object-oriented design. I see a benefit in not leaking any of it into `LPRewards` though I absolutely do agree deploying a separate contract is quite unwieldy. Are you OK with leaving it as-is for now and reconsidering the design for the next escrow implementation (if we ever do that, hopefully, we'll not have to, heh)?

Reply:
> Yeah, you're right. So transfer between escrows requires a beneficiary contract deployed.

And also we would need to `setRewardDistribution(beneficiary)` if I understand the flow correctly.

Reply:
`setRewardDistribution` in this flow would be the escrow directly, not the beneficiary.